pyproject.toml

```diff
@@ -46,6 +46,7 @@ ipython = "^8.21.0"
 jsonschema = "^4.21.1"
 keyring = "*"
 pydantic = "^2.6.1"
+pydantic-partial = "^0.5.5"
```

Comment:
I'm not fully up to speed here, so can I get more detail on why we're creating exceptions to model validations? That seems confusing to me.

If certain model values are optional, it seems like we could communicate that fact in the models themselves by flagging them as optional.

Additionally, if we want to have different scenarios, like "type-only" validation vs. value validation, you might be able to do that with subclassing.

Reply:
@csande The idea behind this was that pydantic expects a fully populated object instance. Doing something like `Task(title="My Title")` will fail because all of the required fields are not supplied. In the case of an update where we only want to update a single field, models needed to be populated and serialized into json with only the fields to be included in the update. This library allows the creation of partial pydantic objects instances.

After some discussion with AD/Michela, I am going to remove this dependency from this PR. Michela is taking care of this problem in here PR [here](https://github.com/canvas-medical/canvas-plugins/pull/68).

Reply:
Removed this dependency.